### PR TITLE
Bumped scylla manager backend to 2022 and removed unsupported versions

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: configurations/manager/debian10.yaml [hunk 1]
```diff
@@ -1,2 +1,4 @@
 ami_id_monitor: 'ami-07d02ee1eeb0c996c'  # Debian buster image - debian-10-amd64-20210208-542
 ami_monitor_user: 'admin'
+
+manager_scylla_backend_version: '2021'  # Notice: Uses 2021.1, while other (newer deb) use 2022, since we support both
```
Comment (fgelcer) [inline configurations/manager/debian10.yaml:4]: understand what is written in the comment, but it should be in a different commit? or to change the commit message to be something like "bump Scylla backend version for manager jobs"

File: configurations/manager/ubuntu18.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 ami_id_monitor: 'ami-08b314ce48a790a19'  # Canonical, Ubuntu, 18.04 LTS, amd64 bionic image build on 2019-07-22
 ami_monitor_user: 'ubuntu'
 
-manager_scylla_backend_version: '2020'  # Notice: Uses 2020.1, while other (newer deb) use 2021, since we support both
+manager_scylla_backend_version: '2021'  # Notice: Uses 2021.1, while other (newer deb) use 2022, since we support both
```
Comment (fgelcer) [inline configurations/manager/ubuntu18.yaml:4]: same here

**Review by fgelcer — COMMENTED** [review]
other than a comment, LGTM

**Review by rayakurl — APPROVED** [review]
Approved. But you need to make sure that the unitests failed due to the known issue today and not something else

**Review by fgelcer — DISMISSED** [review]
LGTM

**Comment by ShlomiBalalis** [comment]
The job I ran failed, but it seems to be a manager issue: https://github.com/scylladb/scylla-manager/issues/3215

**Comment by ShlomiBalalis** [comment]
> Approved. But you need to make sure that the unitests failed due to the known issue today and not something else

@rayakurl There was a test failure from this pr, I fixed it in the latest commit

File: unit_tests/test_sdcm_mgmt_common.py [hunk 3]
```diff
@@ -10,6 +10,6 @@ def test_get_manager_scylla_backend_returns_repo_address(self):  # pylint: disab
         assert url == 'http://downloads.scylladb.com/deb/ubuntu/scylla-2021.1.list'
 
     def test_get_manager_repo_from_defaults_returns_repo_address(self):  # pylint: disable=no-self-use
-        url = get_manager_repo_from_defaults("2.5", Distro.UBUNTU20)
+        url = get_manager_repo_from_defaults("3.0", Distro.UBUNTU20)
 
-        assert url == 'http://downloads.scylladb.com/deb/ubuntu/scylladb-manager-2.5-focal.list'
+        assert url == 'http://downloads.scylladb.com.s3.amazonaws.com/deb/ubuntu/scylladb-manager-3.0-focal.list'
```
Comment (fgelcer) [inline unit_tests/test_sdcm_mgmt_common.py:15]: the URL without `.s3.amazonaws.com` works the same as with it

**Review by fgelcer — APPROVED** [review]
LGTM

File: unit_tests/test_sdcm_mgmt_common.py (hunk 3 above)
Comment (ShlomiBalalis) [reply unit_tests/test_sdcm_mgmt_common.py:15]: I thought so, too. But it's actually not. Apparentely, according to israel, not using the full address can cause connection issues.
Plus, the link in the yaml literally uses the full address, so I kind of have too.

**Review by rayakurl — APPROVED** [review]
approved
